saskatoon/harvest/views.py

```diff
@@ -277,3 +276,69 @@ def harvest_yield_create(request):
             messages.success(request, _("New Fruit Recipient successfully added!"))
 
         return HttpResponseRedirect(request.META.get('HTTP_REFERER'))
+
+
+@login_required
+def harvest_adopt(request, id):
+    """
+    Adds harvest pickleader and changes status to Adopted
+    Used in a button at harvest-detail/status template
+    """
+    harvest = get_object_or_404(Harvest, id=id)
+    current_user = request.user
+
+    if harvest.pick_leader == None:
+        harvest.pick_leader = current_user
+        harvest.status = 'Adopted'
+        harvest.save()
+        messages.success(request, _("You Adopted the Harvest!"))
+    else:
+        messages.warning(request, _("The Harvest already has a pick leader!"))
+
+    return HttpResponseRedirect(request.META.get('HTTP_REFERER'))
+
+
+@login_required
+def harvest_leave(request, id):
+    """
+    Removes harvest pickleader and changes status to Orphan
+    Used in a button at harvest-detail/status template
+    """
+    harvest = get_object_or_404(Harvest, id=id)
+    current_user = request.user
+
+    if harvest.pick_leader == current_user:
+        harvest.pick_leader = None
+        harvest.status = 'Orphan'
+        harvest.save()
+        messages.success(request, _("You Left the Harvest!"))
```

Comment:
I'd suggest we don't capitalize "Left", e.g. : "You successfully left this harvest!", no big deal.

Reply:
I have changed the messages 👍🏿